services/provider/status/status.go

```diff
@@ -0,0 +1,34 @@
+// package status contains the information that will be received from
+// storage client and gets updated at storageconsumer status section
+package status
+
+import (
+	"github.com/hashicorp/go-version"
+)
+
+type Fields struct {
+	ClusterVersion  string
+	OperatorVersion string
+}
+
```

Comment:
Instead of having the func type and creating functions should not we create the func over struct type to assign the values?

```package status

import (
	"github.com/hashicorp/go-version"
)

type StatusInfo struct {
	ClusterVersion  string
	OperatorVersion string
}

func (s *StatusInfo) SetClusterVersion(v string) error {
	if _, err := version.NewSemver(v); err != nil {
		return err
	}
	s.ClusterVersion = v
	return nil
}

func (s *StatusInfo) SetOperatorVersion(v string) error {
	if _, err := version.NewSemver(v); err != nil {
		return err
	}
	s.OperatorVersion = v
	return nil
}```

Reply:
- The requirement is, values in this struct should be optionally set in a function call and backward compatible
- So I had to choose b/n two patterns, functional options vs builder and I chose the former, seems more suggested with gRPC comms
- Considering [this](https://github.com/red-hat-storage/ocs-client-operator/blob/76e5851247/service/status-report/main.go#L90-L92) func, any other params should be optional as not all clients can be on same version
``` go

	if _, err = providerClient.ReportStatus(ctx, storageClient.Status.ConsumerID); err != nil {
		klog.Exitf("Failed to update lastHeartbeat of storageClient %v: %v", storageClient.Status.ConsumerID, err)
	}
```
with func opts I can do 
``` go
	if _, err = providerClient.ReportStatus(ctx, storageClient.Status.ConsumerID,
		status.AddClusterVersion(cVersion),
		status.AddOperatorVersion(opVersion)); err != nil {
		klog.Exitf("Failed to report status of storageClient %v: %v", storageClient.Status.ConsumerID, err)
	}
```
for 
``` go
func (cc *OCSProviderClient) ReportStatus(ctx context.Context, consumerUUID string, fields ...status.Field) (*pb.ReportStatusResponse, error) {
```
with no change to func signature and is forward/backward compatible